File: Calculator/ios-native/CalculatorHub/Calculators/BMICalculator.swift
import Foundation

struct BMIResult {
    let bmi: Double
    let category: String
}

class BMICalculator {
    func calculate(height: Double, weight: Double, isMetric: Bool) -> BMIResult {
        let bmi: Double
        
        if isMetric {
            // Metric: kg / (m^2)
            let heightInMeters = height / 100.0 // cm to m
            bmi = weight / pow(heightInMeters, 2)
        } else {
            // Imperial: 703 * lbs / (in^2)
            bmi = weight * 703.0 / pow(height, 2)
        }
        
        let category: String
        if bmi < 18.5 {
            category = "Underweight"
        } else if bmi < 25 {
            category = "Normal"
        } else if bmi < 30 {
            category = "Overweight"
        } else {
            category = "Obese"
        }
        
        return BMIResult(bmi: bmi, category: category)
    }
    
    func calculateImperial(feet: Double, inches: Double, weight: Double) -> BMIResult {
        let totalInches = (feet * 12) + inches
        return calculate(height: totalInches, weight: weight, isMetric: false)
    }
}

File: Calculator/android-native/app/src/main/java/com/qaantum/calculatorhub/calculators/BMICalculator.kt
package com.qaantum.calculatorhub.calculators

import kotlin.math.pow

data class BMIResult(
    val bmi: Double,
    val category: String
)

class BMICalculator {
    fun calculate(
        height: Double,
        weight: Double,
        isMetric: Boolean
    ): BMIResult {
        val bmi = if (isMetric) {
            // Metric: kg / (m^2)
            val heightInMeters = height / 100.0 // cm to m
            weight / heightInMeters.pow(2)
        } else {
            // Imperial: 703 * lbs / (in^2)
            weight * 703.0 / height.pow(2)
        }

        val category = when {
            bmi < 18.5 -> "Underweight"
            bmi < 25 -> "Normal"
            bmi < 30 -> "Overweight"
            else -> "Obese"
        }

        return BMIResult(bmi, category)
    }

    fun calculateImperial(
        feet: Double,
        inches: Double,
        weight: Double
    ): BMIResult {
        val totalInches = (feet * 12) + inches
        return calculate(totalInches, weight, isMetric = false)
    }
}

